lib/puppet/type/rds_instance.rb

```diff
@@ -0,0 +1,184 @@
+Puppet::Type.newtype(:rds_instance) do
+  @doc = 'type representing an RDS instance'
+
+  ensurable do
+    defaultvalues
+    aliasvalue(:available, :present)
+    newvalue(:destroy) do
+      provider.destroy
+    end
+  end
+
+  newparam(:name, namevar: true) do
+    desc 'the name of the db instance (also known as the db_instance_identifier)'
+    validate do |value|
+    end
+  end
+
+  newparam(:db_name) do
+    desc 'The meaning of this parameter differs according to the database engine you use.
+Type: String
+
+MySQL
+
+The name of the database to create when the DB instance is created. If this parameter is not specified, no database is created in the DB instance.
+
+Constraints:
+
+Must contain 1 to 64 alphanumeric characters
+Cannot be a word reserved by the specified database engine
+PostgreSQL
+
+The name of the database to create when the DB instance is created. If this parameter is not specified, no database is created in the DB instance.
+
+Constraints:
+
+Must contain 1 to 63 alphanumeric characters
+Must begin with a letter or an underscore. Subsequent characters can be letters, underscores, or digits (0-9).
+Cannot be a word reserved by the specified database engine
+Oracle
+
+The Oracle System ID (SID) of the created DB instance.
+
+Default: ORCL
+
+Constraints:
+
+Cannot be longer than 8 characters
+SQL Server
+
+Not applicable. Must be null.'
+    validate do |value|
+    end
+  end
+
+  newparam(:security_groups, :array_matching => :all) do
+    desc 'the security groups to associate the instance'
+  end
+
+  newproperty(:tags) do
+    desc 'the tags for the instance'
+  end
+
+  newproperty(:region) do
+    desc 'the region in which to launch the instance'
+    validate do |value|
+      fail 'region should not contain spaces' if value =~ /\s/
+    end
+  end
+
+  newproperty(:db_instance_class) do
+    desc 'the type to use for the instance (mysql, | postgres)'
+    validate do |value|
+      fail 'db_instance_class should not contain spaces' if value =~ /\s/
+      fail 'db_instance_class should not be blank' if value == ''
+    end
+  end
+
+  newproperty(:availability_zone_name) do
```

Comment:
I think based on other types this should just be `availability_zone` https://github.com/puppetlabs/puppetlabs-aws/blob/master/lib/puppet/type/ec2_instance.rb#L76